File: SvnDiffBrowser/src/main/java/jp/hitting/svn_diff_browser/model/RepositoryModel.java
package jp.hitting.svn_diff_browser.model;

/**
 * Repository Information.
 * Created by hitting on 2016/02/06.
 */
public class RepositoryModel {

    private String url;
    private String userId;
    private String password;

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

}

File: SvnDiffBrowser/src/main/kotlin/jp/hitting/svn_diff_browser/model/RepositoryModel.kt
package jp.hitting.svn_diff_browser.model;

import org.springframework.util.StringUtils

/**
 * Repository Information.
 * Created by hitting on 2016/02/06.
 */
class RepositoryModel {

    var url = ""
    var userId = ""
    var password = ""

    fun getRepositoryName(): String {
        if (StringUtils.isEmpty(this.url)) {
            return ""
        }
        val temp = this.url.split("/")
        return temp.get(temp.size - 1)
    }

    override
    fun hashCode(): Int {
        return (this.url + "^" + this.userId).hashCode()
    }

}
